docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -408,7 +411,7 @@ This section lists out the behavior for Cluster objects using `ClusterClass` in
        ```
     1. For the ControlPlane object in `cluster.spec.topology.controlPlane`
     1. Initializes a control plane object using the control plane template defined in the `ClusterClass.spec.controlPlane.ref field`. Use the name `<cluster-name>`.
-    1. Sets the number of replicas on the control plane object from `spec.topology.controlPlane.replicas`.
+    1. If `spec.topology.controlPlane.replicas` is set, set the number of replicas on the control plane object to that value.
```

Comment:
```suggestion
    1. Checks if the control plane object implements support for the Replicas field. If yes, sets the number of replicas on the control plane object from `spec.topology.controlPlane.replicas`, else ignore the field.
```

Reply:
How can we check if a control plane provider supports this field? (afaik it doesn't have to be set in e.g. the KubeadmControlPlaneTemplate) (I know we could inspect the CRD, but I hope we don't want to do that)
Related to: https://github.com/kubernetes-sigs/cluster-api/pull/4993/files#r675694890

Reply:
We are not checking if the provider support this field, but we are going to act according to what is specified in topology.replica

> // If the value is nil, the ControlPlane object is created without the number of Replicas  
> // and it's assumed that the control plane controller does not implement support for this field.

Reply:
Ignoring unsupported fields (which are not mandatory in the contract) makes the end-user UX a bit easier. And maybe adding a logger or an event (though this legitimately might be overkill) to surface the fact to the end-user is also something we could look into.

Reply:
For now the current text is actually what it can be done technically in the first iteration. @srm09 We should open an issue to make it better later, right now it's probably not something we can do easily.

Reply:
Opened https://github.com/kubernetes-sigs/cluster-api/issues/5019 to track this discussion